### PR TITLE
Add hidden title column to catalog and API table to enable filtering by title

### PR DESCRIPTION
**Pull request by dweber019**
## Hey, I just made a Pull Request!
Fixes https://github.com/backstage/backstage/issues/12263 and https://github.com/backstage/backstage/issues/10418

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by dweber019** [comment]
FYI @freben

**Review by benjdlambert — COMMENTED** [review]
Looks good to me, just one thing about the changeset.

**Comment by benjdlambert** [comment]
I went ahead and fixed this PR up, I'm wondering if this might be needed in the long run with the work that @vinzscam is currently working on?

File: .changeset/silly-geese-design.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-api-docs': patch
+'@backstage/plugin-catalog': patch
```
[the remark this answers is not in the document]
Comment (benjdlambert) [reply .changeset/silly-geese-design.md]: Yeah - I went and fixed in https://github.com/backstage/backstage/pull/12275/commits/b48e19730c0cfed721121dfb6ab4e231f87321f9

[the remark this answers is not in the document]
Comment (benjdlambert) [reply .changeset/silly-geese-design.md]: Ahh, nevermind, I see you also did it too :joy:

[the remark this answers is not in the document]
Comment (dweber019) [reply .changeset/silly-geese-design.md]: Hehe did the same 😉

**Review by benjdlambert — APPROVED** [review]

File: plugins/catalog/src/components/CatalogTable/CatalogTable.tsx [hunk 2]
```diff
@@ -72,6 +72,7 @@ export const CatalogTable = (props: CatalogTableProps) => {
       ...createEntitySpecificColumns(),
       columnFactories.createMetadataDescriptionColumn(),
       columnFactories.createTagsColumn(),
+      columnFactories.createTitleColumn(),
```
Comment (freben) [inline plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:75]: I'd say from an end user perspective, it's probably pretty unexpected to for example go in their code editor and getting nice tab completion for `columnFactories.createTitleColumn()` but then struggling with it just not showing up for unknown reasons. :)

Would you mind adding an `options?: { hidden?: boolean }` to this function and making sure to set that to true in the other files above? That would leave the "expected" use case intact.

Comment (dweber019) [reply plugins/catalog/src/components/CatalogTable/CatalogTable.tsx]: Makes sense, changed the order of the columns too, as it makes more sense to have title before name.

**Review by freben — APPROVED** [review]

**Pull request closed without merging**